### PR TITLE
[CI] Update the PR template

### PR DESCRIPTION
**Pull request by z9r5**
## Description
Add a checkbox about unit tests and how one can check the changes in the PR template.

Signed-off-by: Artem Kladov <artem.kladov@flant.com>

## Why do we need it, and what problem does it solve?
The updates will help not forget the need to check the code and write tests.

## Changelog entries
```changes
section: ci
type: feature
summary: Add a checkbox about unit tests and how one can check the changes.
impact_level: low
```

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -16,6 +16,16 @@
...
 
+## What is the expected result?
+<!---
+  How one can checnk these changes after applying?  
+
+  Describe, what (resource, state, event, etc.) MUST or MUST NOT change/happen after applying these changes.
+-->
+
+## Checklist
+- [ ] The code is covered by unit tests (hooks, templates, matrix)
```
Comment (nabokihms) [inline .github/pull_request_template.md:27]:
```suggestion
- [ ] The code is covered by unit tests (hooks, templates, matrix)
```
This should be split into many checkboxes, and more checkboxes should be added.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -16,6 +16,16 @@
   If it fixes an obvious bug, please tell users about the impact and effect of the problem.
 -->
 
+## What is the expected result?
+<!---
+  How one can checnk these changes after applying?
```
Comment (nabokihms) [inline .github/pull_request_template.md:21]:
```suggestion
  How can one check these changes after applying?  
```

File: .github/pull_request_template.md (hunk 1 above)
Comment (z9r5) [reply .github/pull_request_template.md]: - Splitted into many checkboxes.
- Added more checkboxes (docs and manual test ina cluster)

@nabokihms

Comment (nabokihms) [reply .github/pull_request_template.md]: I think there should be more, e.g., e2e tests are passed.

Comment (z9r5) [reply .github/pull_request_template.md]: Could you elaborate?  Add e2e and what else?

Comment (nabokihms) [reply .github/pull_request_template.md]: As for me, that's all.

Comment (z9r5) [reply .github/pull_request_template.md]: Made changes.
@nabokihms

**Review by nabokihms — APPROVED** [review]

**Pull request merged**